jina/peapods/pods/__init__.py

```diff
@@ -84,7 +83,7 @@ def _fix_exception_context(new_exc, old_exc):
 
 
 class BasePod(ExitFIFO):
```

Comment:
I think we can safely use `ExitStack` since the order does not matter anymore? I would put the `workers` at the top of the stack however